webapi/src/main/scala/org/knora/webapi/routing/HealthRouteWithZIOHttp.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.routing
+
+import spray.json.JsObject
+import spray.json.JsString
+import zhttp.http._
+import zio._
+
+import org.knora.webapi.core.State
+import org.knora.webapi.core.domain.AppState
+
+/**
+ * Provides health check logic
+ */
+trait HealthCheckWithZIOHttp {
+
+  protected def healthCheck(state: State): UIO[Response] =
+    for {
+      _        <- ZIO.logDebug("get application state")
+      state    <- state.getAppState
+      result   <- setHealthState(state)
+      _        <- ZIO.logDebug("set health state")
+      response <- createResponse(result)
+      _        <- ZIO.logDebug("getting application state done")
+    } yield response
+
+  private def setHealthState(state: AppState): UIO[HealthCheckResult] =
```

Comment:
the naming of this function is confusing (to me). `setXY` to me suggests changing a state (e.g. updating a Ref) but what it does is actually turning one representation of a health state into a different representation

Reply:
I totally agree. We need to decide how far we go when refactoring the routes: only shallow refactoring in order for the route to work or deep refactoring / rethinking them (including naming, structure, style etc.)? @subotic what is your opinion on that?

In the meantime I added some docstrings to explain what the methods do.